streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1021,39 +1027,45 @@ public String toString() {
         final Sensor taskCreationSensor;
         final Sensor taskDestructionSensor;
 
-        public StreamsMetricsImpl(Metrics metrics) {
+        public StreamsMetricsImpl(Metrics metrics, String metricGrpName, String sensorNamePrefix,
```

Comment:
Why we want to pass in these as parameters instead of computing them internally? It seems `StreamsMetricsImpl` is still only used once.